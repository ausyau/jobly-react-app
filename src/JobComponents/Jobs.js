/* component which renders a list of individual job card components */

import React, { Component } from 'react';
import JoblyApi from '../JoblyApi';
import { Redirect } from 'react-router-dom';
import UserContext from '../HelperComponents/userContext';
import Jobcard from './Jobcard';
import SearchForm from '../HelperComponents/SearchForm';

export default class Jobs extends Component {
  static contextType = UserContext;

  constructor(props) {
    super(props);
    this.state = {
      jobs: []
    };
    this.getJobs = this.getJobs.bind(this);
  }

  async componentDidMount() {
    if(this.context.username) {
      let result = await JoblyApi.getJobs();
      this.setState({ jobs: result });
    }
  }

  componentWillUnmount() {
    this.setState({ companies: [] });
  }

  async getJobs(search) {
    let result = await JoblyApi.getJobs(search);
    this.setState({ jobs: result });
  }

  render() {

    let loggedIn = <div>
      <SearchForm id='job' search={this.getJobs} />
      {this.state.jobs.map(job => (<Jobcard key={job.id} {...job} applyToJob={this.props.applyToJob} user={this.props.user} />))}
    </div>;

    const currentUser = this.context;

    return (
      <div className="Home">
        {currentUser.username
          ? <div>{loggedIn} </div>
          : <Redirect to='/' />
        }
      </div>
    );
  }
}
