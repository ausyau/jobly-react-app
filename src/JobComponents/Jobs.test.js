import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Jobs from "./Jobs";

it("renders without crashing", function () {
  shallow(<Jobs />);
});

it("matches snapshot with no jobs", function () {
  let wrapper = shallow(<Jobs />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});