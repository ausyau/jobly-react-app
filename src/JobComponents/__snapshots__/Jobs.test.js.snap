// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with no jobs 1`] = `
<div>
  <SearchForm
    id="job"
    search={[Function]}
  />
</div>
`;
